### PR TITLE
HPA : Enhance error message to capture POD details

### LINKED ISSUE #79365: Warning FailedGetResourceMetric horizontal-pod-autoscaler missing request for cpu
**What happened**:

HPA always has a target of `<unknown>/70%` and events that say:

```
Events:
  Type     Reason                        Age                    From                       Message
  ----     ------                        ----                   ----                       -------
  Warning  FailedComputeMetricsReplicas  36m (x12 over 38m)     horizontal-pod-autoscaler  failed to get cpu utilization: missing request for cpu
  Warning  FailedGetResourceMetric       3m50s (x136 over 38m)  horizontal-pod-autoscaler  missing request for cpu
```

- There is a single container in the pods and it has resource requests and limits set.
- The metrics-server is running
- All pods have metrics show in `kubectl top pod`
- All pods have metrics in `kubectl get --raw "/apis/metrics.k8s.io/v1beta1/pods"`

Here's the HPA in YAML:

```yaml
apiVersion: autoscaling/v1
  kind: HorizontalPodAutoscaler
  metadata:
    annotations:
      autoscaling.alpha.kubernetes.io/conditions: '[{"type":"AbleToScale","status":"True","lastTransitionTime":"2019-06-25T09:56:21Z","reason":"SucceededGetScale","message":"the
        HPA controller was able to get the target''s current scale"},{"type":"ScalingActive","status":"False","lastTransitionTime":"2019-06-25T09:56:21Z","reason":"FailedGetResourceMetric","message":"the
        HPA was unable to compute the replica count: missing request for cpu"}]'
    creationTimestamp: "2019-06-25T09:56:06Z"
    labels:
      app: restaurant-monitor
      env: prd01
      grafana: saFkkx6ik
      rps_region: eu01
      team: vendor
    name: myapp
    namespace: default
    resourceVersion: "56108423"
    selfLink: /apis/autoscaling/v1/namespaces/default/horizontalpodautoscalers/myapp
    uid: 7345f8fb-972f-11e9-935d-02a07544d854
  spec:
    maxReplicas: 25
    minReplicas: 14
    scaleTargetRef:
      apiVersion: extensions/v1beta1
      kind: Deployment
      name: myapp
    targetCPUUtilizationPercentage: 70
  status:
    currentReplicas: 15
    desiredReplicas: 0
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

**What you expected to happen**:

No `<unknown>` in HPA target

**How to reproduce it (as minimally and precisely as possible)**:

I can't be sure. It's only a single HPA in our cluster. 10 other HPAs are working OK.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.12.6
- Cloud provider or hardware configuration: EKS

### PR DESCRIPTION
**Pull request by abhijit-dev82**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
As per the discussion in #79365 , the error message needs to provide POD and container details on failing to scale a new POD.

#### Which issue(s) this PR fixes:
Fixes #79365 

#### Does this PR introduce a user-facing change?
Yes

#### Does this PR introduce a user-facing change?
```release-note
NONE
```
#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

### CODE REVIEW

File: pkg/controller/podautoscaler/replica_calculator.go [hunk 1]
```diff
@@ -430,7 +430,7 @@ func calculatePodRequests(pods []*v1.Pod, container string, resource v1.Resource
 				if containerRequest, ok := c.Resources.Requests[resource]; ok {
 					podSum += containerRequest.MilliValue()
 				} else {
-					return nil, fmt.Errorf("missing request for %s", resource)
+					return nil, fmt.Errorf("Container %s in POD %s missing request for %s", c.Name, pod.ObjectMeta.Name, resource)
```
Comment (pbetkier) [inline pkg/controller/podautoscaler/replica_calculator.go:433]: A couple of things:

1) rename POD to Pod, as it's officially spelled [doc](https://kubernetes.io/docs/concepts/workloads/pods/)
2) don't capitalize the error message ([reason](https://github.com/golang/go/wiki/CodeReviewComments#error-strings))
3) nit: I would reword into "missing request for ... in container ... of pod ..." so that the most important info (cause) comes first before container/pod context (where).

**Review by pbetkier — CHANGES_REQUESTED** [review]

File: pkg/controller/podautoscaler/replica_calculator.go (hunk 1 above)
Comment (abhijit-dev82) [reply pkg/controller/podautoscaler/replica_calculator.go:433]: Updated the diffs as per the comments.

**Comment by pbetkier** [comment]
@abhijit-dev82 note you should've done rebase instead of merge. Also, we need an empty release note entry in the description (it was in the PR template) to pass the PR validation.

The code looks good though /lgtm

**Comment by abhijit-dev82** [comment]
Have updated the comment.

**Comment by pbetkier** [comment]
Merges to feature branches are not allowed, you need to do a rebase (produces a cleaner history). I this this should work:

```
# on your branch
$ git reset --hard 2053b8854ac7daec050615ff558b3cb9c0387ca3
$ git rebase master
$ git push -f
```

**Comment by abhijit-dev82** [comment]
Done. Can I squash the commits too?

**Comment by pbetkier** [comment]
> Done. Can I squash the commits too?

yes you can

**Review by pbetkier — APPROVED** [review]
/lgtm

**Review by mwielgus — APPROVED** [review]
/lgtm
/approve
